pkg/router/router.go

```diff
@@ -164,152 +267,80 @@ func registerPropertiesEndpoint(r endpoints.Resource, container *restful.Contain
-		extensionWebService.Route(extensionWebService.POST(fullPath + "/{var:*}").To(ext.handleExtension))
-		extensionWebService.Route(extensionWebService.PUT(fullPath + "/{var:*}").To(ext.handleExtension))
-		extensionWebService.Route(extensionWebService.DELETE(fullPath + "/{var:*}").To(ext.handleExtension))
-	}
-	uidExtensionMap[string(extensionService.UID)] = ext
-}
-
-// Should be called PRIOR to registration of extensionService on informer update
-func UnregisterExtension(extensionService *corev1.Service) {
-	logging.Log.Infof("Removing extension %s", extensionService.Name)
-	serviceRootPath := fmt.Sprintf("%s/%s", extensionWebService.RootPath(), extensionService.Name)
-	extensionMutex.Lock()
-	defer extensionMutex.Unlock()
-	// Grab endpoints to remove from service
-	paths := getExtensionEndpoints(extensionService.ObjectMeta.Annotations[ExtensionUrlKey])
-	for _, path := range paths {
-		fullPath := strings.TrimSuffix(fmt.Sprintf("%s/%s", serviceRootPath, path), "/")
-		// Routes must be removed individually and should correspond to the above registration
-		extensionWebService.RemoveRoute(fullPath, "GET")
-		extensionWebService.RemoveRoute(fullPath, "POST")
-		extensionWebService.RemoveRoute(fullPath, "PUT")
-		extensionWebService.RemoveRoute(fullPath, "DELETE")
-		extensionWebService.RemoveRoute(fullPath+"/{var:*}", "GET")
-		extensionWebService.RemoveRoute(fullPath+"/{var:*}", "POST")
-		extensionWebService.RemoveRoute(fullPath+"/{var:*}", "PUT")
-		extensionWebService.RemoveRoute(fullPath+"/{var:*}", "DELETE")
-	}
-	delete(uidExtensionMap, string(extensionService.UID))
-}
-
-// HandleExtension - this routes request to the extension service
-func (ext Extension) handleExtension(request *restful.Request, response *restful.Response) {
-	target, err := url.Parse("http://" + ext.URL + ":" + ext.Port)
-	if err != nil {
-		utils.RespondError(response, err, http.StatusInternalServerError)
-		return
-	}
-	logging.Log.Debugf("Request Path: %s %+v", request.Request.Method, request.Request.URL.Path)
-	request.Request.URL.Path = strings.TrimPrefix(request.Request.URL.Path, fmt.Sprintf("%s/%s", ExtensionRoot, ext.Name))
-	// Explicitly route to root, better visibility in logs
-	if request.Request.URL.Path == "" {
-		request.Request.URL.Path = "/"
-	}
-	logging.Log.Debugf("Proxy   Path: %s %+v", request.Request.Method, request.Request.URL.Path)
-	proxy := httputil.NewSingleHostReverseProxy(target)
-	proxy.ServeHTTP(response, request.Request)
+// extensionPath constructs the extension path (excluding the root) used by
+// restful.Route
+func extensionPath(extName, path string) string {
+	return strings.TrimSuffix(fmt.Sprintf("%s/%s", extName, path), "/")
 }
 
-// Returns target port if exists, else source == target
-func getServicePort(svc corev1.Service) string {
+// getServicePort returns the target port if exists or the source port otherwise
+func getServicePort(svc *corev1.Service) string {
```

Comment:
Why's this now a *?

Reply:
All services are passed by the informer where they are guaranteed as being non-nil once passed to the router package. Since the structs are relatively big
```
type Service struct {
	metav1.TypeMeta `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
	Spec ServiceSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
	Status ServiceStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
}
```

I changed it to a pointer so that the data wouldn't be copied.

Reply:
👍